### PR TITLE
feat(code-review): PLN-727 Phase 1 — /fix category-dispatch (v2.13.0)

### PR DESCRIPTION
**Pull request by wongk**
## Summary

PLN-727 Phase 1 restructures `/code-review:fix` from a uniform verify-and-edit flow into a **category-dispatch system**. The motivating safety regression: shipping PLN-721 (Premise reviewer) introduced findings with categories `/fix` had no special handling for, so a `Premise/Cohesion` finding could in principle be applied as a code edit at the cited file:line — silently destructive. Every Premise verdict now routes to a category-specific **manual-action template** instead.

This is the first plan in PRD-409's queue that addresses a behavior already shipping today; it closes a hatch every other shipped reviewer (PLN-720/721/722/773/779) was widening.

## Dispatch buckets

| Bucket | What goes here | Handler |
|---|---|---|
| **auto-fix** | Correctness, Code Quality, Documentation, Security, safe Hygiene (`ci_artifacts` / `path_leakage` / `gitignore_drift`) | Direct code edit at anchor line, guarded by mandatory `code_snippet` drift check |
| **callsite-fix** | ImpactAnalysis | **Deferred** — routes to manual-surface today with `impact_semantic_change.md` (full `external_impact[]` list rendered). Activates when PLN-726 ships. |
| **specialized-fix** | TestQuality (`missing-coverage` / `weak-assertion` / `mock-faithfulness` / `missing-edge-case`) | **Deferred** — routes to manual-surface today. `bug-locking` and `test-deletion` permanently manual. Activates when PLN-723 ships. |
| **manual-surface** | Premise (4 subcategories), `Hygiene/sensitive_files`, `InjectionAttempt`, `CompanionChange`, Coverage gaps (re-run-or-acknowledge meta-action) | Category-specific template rendered into the final report |

## Behavioral changes

- **Source: `review_result.json` ONLY.** `validate_output.json` fallback removed per PRD-409 zero-release backward-compat policy.
- **Default in non-TTY: print plan + manual-surface report, exit without applying.** `--apply` required for code modification. Removes the prior auto-yes-after-5s timeout footgun. Interactive (TTY) prompts `Y/N` with no timeout.
- **Mandatory `code_snippet`-drift check before any code-modifying fix.** Re-greps the cited file at `line ±3` for the snippet's first line; miss → `STALE_FINDING` tag, skip, route to manual-surface with drift note. Stops `/fix` from clobbering code when the line has shifted, the bug was already fixed, or the line number is stale.
- **Verification subagent (old Step 2) skipped when `verifier_verdict` is non-null.** PLN-722 already audited those findings — re-verifying is redundant LLM spend. `pending_verification[]` entries still get the per-finding verification subagent.
- **New flags**: `--include-medium`, `--include-tentative`, `--include-justified`, `--dry-run`, `--apply`, `--category-only <name>`, `--skip-verification`. `TENTATIVE` and `JUSTIFIED-VALID` are manual-surface only — never opt into auto-fix.
- **`run-loop.sh` updated to invoke `/code-review:fix $cr_dir --apply`** so post-loop fix sessions retain auto-apply behavior under the new default.

## Files

| File | Change |
|---|---|
| `plugins/code-review/skills/fix/SKILL.md` | Full restructure (165 → ~330 lines): 7-step flow + dispatch table |
| `plugins/code-review/skills/fix/templates/*.md` | **13 new** manual-action templates |
| `plugins/code-review/.claude-plugin/plugin.json` | v2.12.1 → **v2.13.0** (MINOR) |
| `plugins/code/scripts/run-loop.sh` | Adds `--apply` to `/fix` invocation |
| `plugins/code/.claude-plugin/plugin.json` | v1.12.2 → **v1.12.3** (PATCH) |
| `CHANGELOG.md` | v2.13.0 + v1.12.3 entries |

## Scope of this PR

PLN-727 was originally written to ship LAST after every other PRD-409 plan. We currently have plans 01/02/03 of its dependencies shipped (PLN-720 / PLN-721 / PLN-722; plus PLN-773 / PLN-774 / PLN-779). This PR scopes to:

- ✅ **Phase A** (source + dispatch foundation — deps: PLN-722 ✅)
- ✅ **Phase D** (manual-surface templates — deps: PLN-720 ✅ + PLN-721 ✅)
- ✅ **Partial Phase E** (coverage-gap manual template — deps: PLN-721 ✅; coverage acks reference PLN-725 which is deferred but the template documents the schema today)

**Deferred safely** (their findings route to manual-surface so no category can apply unsafe edits):

- ⏳ **Phase B** (callsite-fix flow) — needs PLN-726
- ⏳ **Phase C** (specialized TestQuality fix) — needs PLN-723
- ⏳ **Phase E complete** (extract-signals coverage acks gating the verdict) — needs PLN-725

## Test plan

- [x] `pytest plugins/` — **1378 passed, 5 skipped**
- [x] `ruff check plugins/code-review/ plugins/code/` — clean
- [x] `pyright plugins/code-review/tools/python/` — 0 errors
- [x] `bash -n plugins/code/scripts/run-loop.sh` — OK
- [x] Code-fence balance check on `SKILL.md` and all 13 templates — all balanced
- [ ] CI green on push
- [ ] Manual smoke: next run-loop iteration confirms `/fix` invokes with `--apply` and dispatches a Premise finding (if present) to manual-surface

## Coordination with open PRs

- **PR #119** (start.md scope wrapper alignment, v2.12.2) is still open. If PR #119 merges first → main goes to v2.12.2; this PR rebases cleanly (no overlapping files). If this PR merges first → PR #119 needs to bump to v2.13.1 because main will be at v2.13.0.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: plugins/code-review/skills/fix/SKILL.md [hunk 1]
```diff
@@ -19,97 +21,145 @@ $ARGUMENTS
...
+| Category | Subcategory | Bucket | Notes |
+|---|---|---|---|
+| `Correctness` | — | **auto-fix** | Direct code edit at anchor line |
+| `Code Quality` | — | **auto-fix** | DRY / maintainability — auto-fix at anchor |
+| `Documentation` | — | **auto-fix** | Edit cited file:line |
+| `Security` | — | **auto-fix** | Same flow as Correctness |
+| `Hygiene` / `Repo Hygiene` | `ci_artifacts` | **auto-fix** | Delete or gitignore |
+| `Hygiene` / `Repo Hygiene` | `path_leakage` | **auto-fix** | Replace path with env var / config reference |
+| `Hygiene` / `Repo Hygiene` | `gitignore_drift` | **auto-fix** | Add gitignore entry |
+| `Hygiene` / `Repo Hygiene` | `sensitive_files` | **manual-surface** | Never auto-modify (.env, credentials, .pem) |
```
Comment (thadeusb) [inline plugins/code-review/skills/fix/SKILL.md]: This is the row that worries me, because it's the exact silently-destructive case the PR exists to prevent and it doesn't actually fire. The sensitive_files routing keys on a subcategory the hygiene reviewer never sets. Look at cmd_hygiene in code_review_helpers.py: _check_sensitive_files (and ci_artifacts/path_leakage/gitignore_drift) emit category 'Repo Hygiene' with no subcategory, and normalize_finding defaults subcategory to None. Nothing in the pipeline ever assigns it, the final review_result.json fixtures all show subcategory null. So a committed .env or *.pem arrives as Repo Hygiene / subcategory=None, misses this row, and falls through to the 'Hygiene (other/unset) -> auto-fix' row right below. _severity_for_hygiene_file gives .env/.pem/.key HIGH (they're in HIGH_EXTS), so it clears the severity gate too. End result: /fix dispatches a sonnet agent to edit a .env at the anchor line. Either populate subcategory in cmd_hygiene, or flip the Hygiene default to manual-surface and only auto-fix on a positive ci_artifacts/path_leakage/gitignore_drift signal. Same root cause makes the whole subcategory column mostly decorative against today's data, but Hygiene is the only one whose fallback is unsafe.

File: plugins/code-review/skills/fix/SKILL.md [hunk 2]
```diff
@@ -19,97 +21,145 @@ $ARGUMENTS
...
-System-scoped findings (`finding_scope: "system"`) — e.g. coverage gaps with `system_marker: "budget-exceeded"` or `coverage:<reviewer>` — cannot be auto-fixed in code; surface them in the summary as "manual surface" items and skip the verify/fix loop for them.
+Read `<CR_DIR>/review_result.json` (canonical envelope per PLN-719). **There is no legacy fallback** — zero-release backward compat per PRD-409. File missing → error `"No code review output found in CR_DIR. Run a code review first."` → exit 1.
 
-No actionable BLOCKING/HIGH findings → print "No actionable findings. Review complete." → mark all todos completed → exit.
+Findings live in:
 
----
+| Bucket | What it contains | Default action |
+|---|---|---|
+| `envelope.verified[]` | CONFIRMED + DOWNGRADE + TENTATIVE + RE_ASSERTED + JUSTIFIED-INVALID | filter into dispatch |
```
Comment (thadeusb) [inline plugins/code-review/skills/fix/SKILL.md:61]: RE_ASSERTED isn't a real verdict and re-assert isn't a real command. VERIFIER_VERDICTS in code_review_schema.py is {CONFIRMED, DOWNGRADE, TENTATIVE, REJECTED, JUSTIFIED-VALID, JUSTIFIED-INVALID}, and verifier_verdict is enum-validated against it, so a finding written with RE_ASSERTED fails schema validation. There's also no re-assert subparser in code_review_helpers.py and this PR doesn't add one. Every manual-surface template ends with 're-assert via code_review_helpers.py re-assert ...' as the operator's primary recourse, and this row claims verified[] contains RE_ASSERTED. Anyone who follows those templates gets argparse 'invalid choice: re-assert'. Wire up the subcommand and add the verdict, or stop advertising it across 14 files.

File: plugins/code-review/skills/fix/SKILL.md [hunk 3]
```diff
@@ -19,97 +21,145 @@ $ARGUMENTS
...
+| `envelope.justified[]` | JUSTIFIED-VALID (author defended, verifier confirmed) | exclude unless `--include-justified` (manual-surface only) |
+| `envelope.rejected[]` | REJECTED by verifier | always exclude |
+| `envelope.pending_verification[]` | Verifier didn't run on this finding (verifier failed, was deferred past budget, or `--no-verify` bypass) | include if severity matches; always **manual-surface** with the `pending_verification.md` template — `/fix` does NOT re-implement the verifier |
+| `envelope.coverage_gaps[]` | System-scoped coverage findings | always include if severity matches; manual-surface bucket |
 
-## Step 2: Verify Each Finding
+### Apply severity + verdict filter
+
+For each candidate finding (from `verified[]` + `pending_verification[]` + `coverage_gaps[]`):
```
Comment (thadeusb) [inline plugins/code-review/skills/fix/SKILL.md]: --include-justified is a no-op as written. The candidate set here is verified[] + pending_verification[] + coverage_gaps[], justified[] is never pulled in, so the flag has nothing to surface. Add justified[] to the candidate list when the flag is set, or drop the flag.

File: plugins/code-review/skills/fix/SKILL.md [hunk 4]
```diff
@@ -19,97 +21,145 @@ $ARGUMENTS
...
+### 3a. Auto-fix bucket
+
+Group surviving auto-fix findings by file. Apply **sequentially** within file (later fixes see earlier results); files run sequentially overall to keep the run-loop reproducible.
+
+For each finding, perform the mandatory drift check first:
 
-Group confirmed findings by file. Fix **sequentially** (one agent at a time) — findings may share source files, so each fix must see prior results.
+```bash
+# Drift check (inline shell — no helper subcommand needed)
+grep -Fn "<first non-empty line of finding.code_snippet>" "<finding.file>" | \
```
Comment (thadeusb) [inline plugins/code-review/skills/fix/SKILL.md:163]: The mandatory drift check assumes every auto-fix finding carries a code_snippet, but hygiene findings don't emit one and normalize defaults it to empty string. grep -Fn "" matches every line in the file, so the gate is a no-op for the entire Hygiene auto-fix bucket. Stacked on the sensitive_files routing above, that means nothing at all guards a sensitive-file auto-edit.

(Anchored at the Step 3a drift-check command; the review referenced line 116, but the grep -Fn drift check this concerns lives here.)

**Review by thadeusb — CHANGES_REQUESTED** [review]
Two things defeat the safety guarantee this whole PR is built on. Sensitive-file hygiene findings route to auto-fix instead of manual-surface, because the dispatch keys on a subcategory the reviewers never populate, so /fix will happily spawn an agent to edit a committed .env at the cited line. And the re-assert escape hatch every template points operators at isn't a real command or a real verdict. The dispatch keys need to be reconciled against what reviewers actually emit before this lands. Will re-review.

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,40 @@ All notable changes to the claude-plugins project will be documented in this fil
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/). Entries are listed newest-first; each plugin section is treated as released when merged to `main`.
 
+### code-review v2.13.1
+
+#### Added
```
Comment (shafty023) [inline CHANGELOG.md:9]: **🟡 MEDIUM — [P2] Hand-written CHANGELOG documents intermediate versions that never shipped as discrete releases**

CLAUDE.md Learned Patterns (Documentation & Versioning) states: 'Never hand-write CHANGELOG entries for plugin changes; when reviewing generated release notes, verify the version heading matches `plugin.json` and every `Fixed`/`Replaces` claim maps to behavior that existed before the PR.' The diff adds four CHANGELOG sections by hand: `code-review v2.13.1`, `code-review v2.13.0`, `code v1.12.4`, and `code v1.12.3`. The corresponding `plugins/code-review/.claude-plugin/plugin.json` diff shows a single jump from `2.12.2` to `2.13.1`, meaning versions `2.13.0` and `2.12.3` never existed as shipped artifacts. Recording intermediate phantom versions violates the rule that 'every `Fixed`/`Replaces` claim maps to behavior that existed before the PR' — the `v2.13.0 -> v2.13.1` delta (exit-code contract + fix_result.json) did not exist as a shipped version that could be 'fixed' in a subsequent release. The CHANGELOG should reflect the single promoted version (`2.13.1`) rather than a two-stage version fiction.

**Recommendation:** Run `/update-documentation` (as mandated by CLAUDE.md) after finalizing code changes instead of manually writing CHANGELOG entries. If the staged approach is intentional, collapse both `v2.13.0` and `v2.13.1` sections into a single `v2.13.1` entry matching the final `plugin.json` version, and do the same for `code` (`v1.12.4` combining both v1.12.3 and v1.12.4 changes).

File: plugins/code-review/skills/fix/templates/pending_verification.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
+### ⚠️  Pending Verification — `{file}:{line}`  ({severity}/{category}{subcategory})
```
Comment (shafty023) [inline plugins/code-review/skills/fix/templates/pending_verification.md]: **🟡 MEDIUM — [P2] Missing separator between {category} and {subcategory} in pending_verification.md header**

The template header reads `({severity}/{category}{subcategory})`. The two placeholders are concatenated with no delimiter, so a finding with category 'Correctness' and subcategory 'null-deref' renders as `HIGH/Correctnessnull-deref`. Every other template that uses both fields in a display context separates them (e.g., _generic.md uses `{category} / {subcategory}`, testquality_specialized.md uses `{category} / {subcategory}`). The pending_verification template is the only one that omits the separator, producing malformed rendered output for any finding that has both fields populated.

**Recommendation:** Change the header to `({severity}/{category}/{subcategory})` or `({severity}/{category} / {subcategory})` to match the pattern used in the other templates. If subcategory may be absent, the skill rendering logic should conditionally omit it (e.g., render `{severity}/{category}` when subcategory is empty), which is already implied by the SKILL.md note 'Missing placeholder data -> leave the placeholder visible'.

File: plugins/code-review/skills/fix/SKILL.md [hunk 7]
```diff
@@ -19,97 +21,145 @@ $ARGUMENTS
...
+Auto-fix:        N findings  (breakdown by category)
+Callsite-fix:    N findings  (deferred — surfaced as manual)  [appears only if PLN-726 not yet shipped]
+Specialized:     N findings  (deferred — surfaced as manual)  [appears only if PLN-723 not yet shipped]
+Manual surface:  N findings  (breakdown by category/subcategory)
+
+Total auto-action: N findings.
+Total requiring human action: N findings (surfaced in final report).
+
+Interactive mode → Y/N prompt required.
+Non-interactive → requires --apply to modify code; --dry-run for explicit no-op.
```
Comment (shafty023) [inline plugins/code-review/skills/fix/SKILL.md:140]: **🔴 BLOCKING — [P0] validate_output.json fallback removed but start.md never writes review_result.json**

The new SKILL.md requires `<CR_DIR>/review_result.json` as the sole input (line 135: 'There is no legacy fallback') and errors with exit 1 if the file is absent. The producer is `plugins/code-review/commands/start.md`, which runs `python3 <HELPERS> validate ... > <CR_DIR>/validate_output.json` (unchanged in this PR) and never writes `review_result.json`. Grepping the entire repo for `review_result` returns zero matches outside this SKILL.md diff. Every cr-dir produced by the current review pipeline therefore lacks `review_result.json`, so every invocation of `/code-review:fix` after this PR merges will immediately exit 1 with 'No code review output found in CR_DIR.' The old SKILL.md had an explicit fallback: 'Prefer CR_DIR/review_result.json. Fall back to CR_DIR/validate_output.json (legacy).' That fallback was the only working path for all existing callers. Removing it without a coordinated update to start.md (or a helper that converts validate_output.json to review_result.json) makes /fix completely non-functional for the current pipeline.

**Recommendation:** Either (a) keep the validate_output.json fallback in SKILL.md until start.md is updated to produce review_result.json, or (b) add a migration step (e.g., a code_review_helpers.py subcommand that writes review_result.json from validate_output.json) and call it from start.md in this same PR.

**Review by shafty023 — COMMENTED** [review]
## Code Review Summary

**Status:** Changes Requested
**Reviewers:** Bug Hunter A, Bug Hunter B, Unified Auditor, Premise Reviewer, Contract / Compatibility Reviewer, Security / Abuse Reviewer, Validation / E2E Reviewer
**Model Routing:** Medium -- risk-routed lanes; specialty lanes: Contract / Compatibility Reviewer, Security / Abuse Reviewer, Validation / E2E Reviewer

### Findings

| Severity | Count |
|----------|-------|
| Blocking | 1 |
| High | 1 |
| Medium | 2 |

### BLOCKING Issues (must fix)

1. **[P0] [plugins/code-review/skills/fix/SKILL.md:135]** [P0] validate_output.json fallback removed but start.md never writes review_result.json

### HIGH Issues (should fix)

1. **[P1] [plugins/code/scripts/run-loop.sh:1066]** [P1] Exit-2 (manual action required) logged as fix_error before the exit-2 branch fires _(summary-only)_

### MEDIUM Issues (consider)

1. **[P2] [CHANGELOG.md:9]** [P2] Hand-written CHANGELOG documents intermediate versions that never shipped as discrete releases
2. **[P2] [plugins/code-review/skills/fix/templates/pending_verification.md:1]** [P2] Missing separator between {category} and {subcategory} in pending_verification.md header

### Validation Stats

- **Validated:** 4
- **Cross-file grouped:** 0
- **Summary-only out-of-range P0/P1:** 1
- **Stale bot threads queued for resolution:** 0

**Recommendation:** Do not merge until blocking issues are fixed.

File: plugins/code-review/skills/fix/SKILL.md (hunk 7 above)
Comment (wongk) [reply plugins/code-review/skills/fix/SKILL.md:140]: Rejecting this finding with evidence — the reviewer used too narrow a grep.

`review_result.json` IS produced by every full review run:

- **`cmd_finalize_result`** in `code_review_helpers.py:1599` writes `<CR_DIR>/review_result.json` (the canonical Phase 2 envelope).
- Registered as **`stage_25_finalize_result`** in the run plan at `code_review_helpers.py:6580`.
- Documented in **`commands/start.md:293`** verbatim: *"writes `<CR_DIR>/review_result.json` (the canonical envelope) BEFORE running schema validation. PLN-722: prefers `<CR_DIR>/findings_verified.json` (verify-consolidate output) when present and honors its `force_human_review` flag in the verdict computation; falls back to `findings_validated.json` (everything to `verified[]`) when verify-consolidate didn't run."*

A `grep -rn review_result plugins/code-review/tools/python/code_review_helpers.py` returns at least 12 matches. The old SKILL.md's `validate_output.json` fallback was vestigial — PLN-719 Foundation Phase B made `review_result.json` the canonical input for every consumer (`/fix`, `re-assert`, `review-dismissed-prepare`, downstream verdict computation). Removing the fallback in v2.13.2 brings the skill in line with the rest of the post-PLN-719 surface.

No code change; the producer chain is intact. Resolving.
